rust/xaynet-core/src/mask/masking.rs

```diff
@@ -897,4 +1013,135 @@ mod tests {
     test_masking_and_aggregation!(pow_i64_b4, Power2, i64, 10_000, 10, 5);
     test_masking_and_aggregation!(pow_i64_b6, Power2, i64, 1_000_000, 10, 5);
     test_masking_and_aggregation!(pow_i64_bmax, Power2, i64, 10, 5);
+
+    /// Generate tests for masking, aggregation and unmasking of multiple models:
+    /// - generate random scalars from a uniform distribution with a seeded PRNG
+    /// - scale a model of unit weights, mask and aggregate it to the aggregated masked models
+    /// - derive a mask from the mask seed and aggregate it to the aggregated masks
+    /// - unmask the aggregated masked model
+    /// - check that all aggregated unmasked weights are equal to the original unit weights (up
+    ///   to a tolerance determined by the masking configuration)
+    ///
+    /// The arguments to the macro are:
+    /// - a suffix for the test name
+    /// - the group type of the model and scalar (variants of `GroupType`)
+    /// - the data type of the model and scalar (either float primitives or float variants of
+    ///   `DataType`)
+    /// - an absolute bound for the scalar (optional, choices: 1, 100, 10_000, 1_000_000)
+    /// - the number of weights per model
+    /// - the number of models
+    macro_rules! test_masking_and_aggregation_scalar {
```

Comment:
The macro is quite long, and even though the comments are very helpful, perhaps it would be worth breaking it down into smaller pieces, either with other macros, or with functions? Each step could be its own macro/function for example.

Reply:
Although these are tests, so maybe it's not worth the effort.

Reply:
they're tests after all, but there is enough repetition in the masking tests to justify some refactor-tidying, I agree.